src/Microsoft.OpenApi/Writers/SpecialCharacterStringExtensions.cs

```diff
@@ -42,7 +42,8 @@ public static class SpecialCharacterStringExtensions
         // http://www.yaml.org/spec/1.2/spec.html#style/flow/plain
         private static readonly string[] _yamlPlainStringForbiddenCombinations =
         {
-            ": ",
+            //": ",
+            ":",  // Even though colons can be allowed, this change prevents unquoted trailing colon which causes a problem
```

Comment:
Even though colons not followed by a space can be allowed, this change prevents unquoted trailing colon which causes a problem.